pkg/steps/artifacts.go

```diff
@@ -693,6 +694,35 @@ func gatherContainerLogsOutput(podClient PodClient, artifactDir, namespace, podN
 	return kerrors.NewAggregate(validationErrors)
 }
 
+// for gathering successful build logs to the artifacts, there is no way to augment the pod spec
+// created by the build controller to add the artifacts container; this method cherry picks elements
+// from downloadArtifacts and gatherContainerLogsOutput and munges them in conjunction with the build
+// api logging capabilities; also, without needing to inject an artifacts container, some of the complexities
+// around download/copy from the artifacts container's volume mount and multiple pods are avoided.
+func gatherSuccessfulBuildLog(buildClient BuildClient, artifactDir, namespace, buildName string) error {
+	// adding a subdir to the artifactDir path similar to downloadArtifacts adding the container-logs subdir
+	dir := filepath.Join(artifactDir, "openshift-docker-build-strategy-logs")
+	if err := os.MkdirAll(dir, 0750); err != nil {
```

Comment:
Since other steps will call that function too, there is a chance that this directory we already exist, right?

Reply:
ah ... sure, if multiple steps are possible, I should filter out already exist errors ... will update shortly

Reply:
check that ... the godoc/code from os.MkDirAll ... it is benign and just returns if the dir already exists:

```
// If path is already a directory, MkdirAll does nothing
// and returns nil.
func MkdirAll(path string, perm FileMode) error {
	// Fast path: if we can tell whether path is a directory or file, stop with success or error.
	dir, err := Stat(path)
	if err == nil {
		if dir.IsDir() {
			return nil
		}
```

Reply:
Yea `MkdirAll` === `mkdir -p`